### PR TITLE
feat: implement summaries

### PR DESCRIPTION
**Pull request by MDavidson17**
Utilises pystac.Summarizer.summarize() and a custome fields.json url to construct the summaries for Historical Imagery.

### CODE REVIEW

**Comment by amfage** [comment]
Should the LINZ stac_extensions be bumped to v0.0.12 or v0.0.13 to validate requiring summaries fields?

File: topo_processor/util/transfer_collection.py [hunk 1]
```diff
@@ -59,7 +59,11 @@ def transfer_collection(collection: Collection, target: str):
         json_item = stac_item.to_dict(include_self_link=False)
         write_json(json_item, os.path.join(target, item.collection.title, f"{item.id}.json"))
 
+    # after all items have been processed generate summaries
+    collection.generate_summaries(stac_collection)
```
Comment (amfage) [inline topo_processor/util/transfer_collection.py:63]: I think `generate_summaries` needs to be called by the `validate` command as well, as that doesn't call `transfer_collection`.

**Review by amfage — APPROVED** [review]
Neat way to generate the summaries :+1:

**Review by paulfouquet — APPROVED** [review]
Looks good!

**Pull request merged**